mantidproject/mantid
# Refactor to improve naming of lookup table in refl gui

The table on the Experiment tab in the reflectometry GUI was originally referred to as something like "per-theta defaults" or "per angle defaults". This naming is no longer suitable now we want to look up on more criteria than just angle (#31795). The naming is also sometimes unclear as to whether one is dealing with a single row in the table, or the entire table.

This PR renames classes and variable such that a row is now called a `LookupRow` and the whole table (i.e. vector of rows) is the `LookupTable`.

Note that `defaultsForTheta` and `wildcardDefaults` have been consolidated into a single function `findLookupRow` which takes an optional angle to do the lookup on, or does a wildcard row search otherwise.

**To test:**

- Open the ISIS Reflectometry GUI and set instrument to INTER
- In the Runs tab, enter a run and angle, e.g. run `13460` and angle `0.5`
- On the Experiment tab, enter some values in the table, e.g. transmission run `13463`
- On the runs tab click Process
  - If the run angle matches that in the table it should pick up the transmission run and you'll get an extra workspace named TRANS_13463 in the outputs. This won't appear if the row was not matched.
- Clear the workspaces and try again with different criteria in the table, each time checking that the trans workspace does/doesn't appear according to whether you expect the lookup to have worked:
  - try entering lookup criteria with or without an angle
  - try entering an angle that doesn't match (note that there is a tolerance of 0.01 on the angle lookup)
  - try multiple rows with/without angle(s)

Part of #31795

*This does not require release notes* because **it is refactoring only and does not change functionality**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
